packages/itwin/tree-widget/src/components/trees/stateless/models-tree/ModelsTreeComponent.tsx

```diff
@@ -58,21 +59,60 @@ function ModelsTreeComponentImpl({
   selectionStorage,
   ...treeProps
 }: StatelessModelsTreeComponentProps & { iModel: IModelConnection; viewport: ScreenViewport }) {
+  const [error, setError] = useState<StatelessModelsTreeError | undefined>();
   const availableModels = useAvailableModels(iModel);
   const { filter, applyFilter } = useFiltering();
+  const errorRef = useLatest(error);
   const density = treeProps.density;
 
+  useEffect(() => {
+    return selectionStorage.selectionChangeEvent.addListener(() => {
+      if (isInstanceFocusError(errorRef.current)) {
+        setError(undefined);
+      }
+    });
+  }, [selectionStorage, errorRef]);
+
   const onModelsTreeFeatureUsed = (feature: string) => {
+    if (feature === "instancesfocus" && isInstanceFocusError(error)) {
+      setError(undefined);
+    }
```

Comment:
This piece of code seems out of place. What do you think about the following?

1. In `InstanceFocusError`, rename `onFeatureUsed` to `onInstanceFocusDisabled`. Remove the `feature` argument from that function.

2. When rendering `InstanceFocusError`, pass a function like this:
  ```tsx
  const onInstanceFocusDisabled = () => {
    isInstanceFocusError(error) && setError(undefined);
    onModelsTreeFeatureUsed("instancesfocus");
  };
  ```

Reply:
Found out that the state can be reset when filtered paths are being calculated which simplifies the implementation quite a bit and so this is no longer necessary.